### PR TITLE
[component specs] Cross link from component spec index pages to set up pages

### LINKED ISSUE #3657: Improved the component reference linking
In each of the component references this should point at the start to the component definition

For example this https://docs.dapr.io/reference/components-reference/supported-state-stores/ should include a link to https://docs.dapr.io/operations/components/setup-state-store/ at the start of the topic
And https://docs.dapr.io/reference/components-reference/supported-pubsub/ should include a link to 
https://docs.dapr.io/operations/components/setup-pubsub/

Same applies to
-  https://docs.dapr.io/operations/components/setup-secret-store/
- https://docs.dapr.io/operations/components/setup-bindings/
- https://docs.dapr.io/operations/components/middleware/

Then specifically in the https://docs.dapr.io/operations/components/setup-pubsub/ link this should include and an mention a link to scoping here https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-scopes/ so that it is understood that this is an option in all pub/sub components.

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Cross link and reorganize partial

## Issue reference

PR will close: #3657

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/_index.md [hunk 1]
```diff
@@ -9,8 +9,10 @@ aliases:
 no_list: true
 ---
 
-Every binding has its own unique set of properties. Click the name link to see the component YAML for each binding.
+The following table lists input and output bindings supported, at various levels, by the Dapr bindings building block. [Learn how to set up different input and output binding components for Dapr bindings.]({{< ref setup-bindings.md >}})
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/_index.md]: What did you mean by "at various levels"? Do you mean certification status? As in 
"The following table lists input and output bindings with that can be used by the bindings building block API with their certification status."

File: daprdocs/content/en/reference/components-reference/supported-bindings/_index.md [hunk 2]
```diff
@@ -9,8 +9,10 @@ aliases:
...
 
 {{< partial "components/description.html" >}}
 
+Every binding has its own unique set of properties. Click the name link to see the component YAML for each binding.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/_index.md]:
```suggestion
Every binding component has its own set of properties. Click the name link to see the component specification for each binding.
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/_index.md [hunk 3]
```diff
@@ -8,6 +8,9 @@ no_list: true
 aliases:
 - /developing-applications/middleware/supported-middleware/
 ---
+
+The following table lists middleware components supported, at various levels, by Dapr. [Learn how to customize processing pipelines and set up middleware components.]({{< ref "middleware.md" >}})
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-middleware/_index.md]: Same comment as above. What did you mean by "at various levels"? Do you mean certification status?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md [hunk 4]
```diff
@@ -9,6 +9,12 @@ aliases:
 no_list: true
 ---
 
+The following table lists publish and subscribe brokers supported, at various levels, by the Dapr pub/sub building block. [Learn how to set up different brokers for Dapr publish and subscribe.]({{< ref setup-pubsub.md >}})
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: Same comment as above. What did you mean by "at various levels"? Do you mean certification status?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md [hunk 5]
```diff
@@ -9,6 +9,12 @@ aliases:
...
+
 {{< partial "components/description.html" >}}
 
+{{% alert title="Note" color="primary" %}}
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: Why add this note here?

File: daprdocs/content/en/reference/components-reference/supported-secret-stores/_index.md [hunk 6]
```diff
@@ -9,6 +9,8 @@ aliases:
 no_list: true
 ---
 
+The following table lists secret stores supported, at various levels, by the Dapr secrets building block. [Learn how to set up different secret stores for Dapr secrets management.]({{< ref setup-secret-store.md >}})
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-secret-stores/_index.md]: Same comment as above. What did you mean by "at various levels"? Do you mean certification status?

File: daprdocs/content/en/reference/components-reference/supported-state-stores/_index.md [hunk 7]
```diff
@@ -9,10 +9,12 @@ aliases:
 no_list: true
 ---
 
-{{< partial "components/description.html" >}}
+The following table lists state stores supported, at various levels, by the Dapr state management building block. [Learn how to set up different state stores for Dapr state management.]({{< ref setup-state-store.md >}})
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/_index.md:12]: Same comment as above. What did you mean by "at various levels"? Do you mean certification status?

File: daprdocs/layouts/partials/components/description.html [hunk 8]
```diff
@@ -1,15 +1,29 @@
...
+    </tr>
+    <tr>
+      <td>Component version</td>
+      <td>The version of the component</td>
+      <td>v1</td>
+    </tr>
+    <tr>
+      <td>Since runtime version</td>
+      <td>The version of the Dapr Runtime in which the component <br> first moved to the current status</td>
+      <td>1.11</td>
```
Comment (msfussell) [inline daprdocs/layouts/partials/components/description.html:27]: What is this 1.11 here for?

File: daprdocs/layouts/partials/components/description.html [hunk 9]
```diff
@@ -1,15 +1,29 @@
...
+      </td>
+    </tr>
+    <tr>
+      <td>Component version</td>
+      <td>The version of the component</td>
+      <td>v1</td>
+    </tr>
+    <tr>
+      <td>Since runtime version</td>
+      <td>The version of the Dapr Runtime in which the component <br> first moved to the current status</td>
```
Comment (msfussell) [inline daprdocs/layouts/partials/components/description.html]: This is clunky wording now. Also it contains a <br> html component, which should not be there IMO. 
 Should say something like this

```suggestion
      <td>The version of the Dapr runtime when the component status was set or updated </td>
```

**Review by msfussell — COMMENTED** [review]
Reviewed

File: daprdocs/content/en/reference/components-reference/supported-bindings/_index.md (hunk 1 above)
Comment (hhunter-ms) [reply daprdocs/content/en/reference/components-reference/supported-bindings/_index.md]: I don't remember, honestly - I'll just remove that phrase.

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md (hunk 5 above)
Comment (hhunter-ms) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: ope, I added to the wrong doc. I'll move.

File: daprdocs/layouts/partials/components/description.html (hunk 8 above)
Comment (hhunter-ms) [reply daprdocs/layouts/partials/components/description.html:27]: the existing table has an 'example' column, so 1.11 is the example of the runtime version that could be seen below

**Review by yaron2 — APPROVED** [review]

**Pull request merged**